includes/rest/class-application-controller.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * Application Controller file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use Activitypub\Model\Application;
+
+/**
+ * ActivityPub Application Controller.
+ */
+class Application_Controller extends \WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'application';
+
+	/**
+	 * Register routes
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves the application actor profile.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 * @return \WP_REST_Response Response object.
+	 */
+	public function get_item( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$json = ( new Application() )->to_array();
+
+		$rest_response = new \WP_REST_Response( $json, 200 );
+		$rest_response->header( 'Content-Type', 'application/activity+json; charset=' . \get_option( 'blog_charset' ) );
+
+		return $rest_response;
+	}
+
+	/**
+	 * Retrieves the schema for the application endpoint.
+	 *
+	 * @return array Schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$this->schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'application',
+			'type'       => 'object',
+			'properties' => array(
+				'@context'                  => array(
+					'type'  => 'array',
+					'items' => array(
+						'type' => array( 'string', 'object' ),
+					),
+				),
+				'id'                        => array(
+					'type'   => 'string',
+					'format' => 'uri',
+				),
+				'type'                      => array(
+					'type' => 'string',
+					'enum' => array( 'Application' ),
+				),
+				'name'                      => array(
+					'type' => 'string',
+				),
+				'icon'                      => array(
```

Comment:
I like the idea, but sadly this is not that easy to verify. Per spec an Icon can be a `Link` or `Image` object or a directly the `URI` as `string` or an array of these: https://www.w3.org/TR/activitystreams-vocabulary/#dfn-icon

Reply:
Thanks! That's exactly the type pf feedback I was looking for. Fixed in 75015c3.